File: core/domain/src/main/java/com/gaugustini/mhfudatabase/domain/filter/ArmorFilter.kt
package com.gaugustini.mhfudatabase.domain.filter

import com.gaugustini.mhfudatabase.domain.enums.EquipmentType
import com.gaugustini.mhfudatabase.domain.enums.Gender
import com.gaugustini.mhfudatabase.domain.enums.HunterType
import com.gaugustini.mhfudatabase.domain.model.Armor

/**
 * Filter for [Armor].
 */
data class ArmorFilter(
    val name: String? = null,
    val type: EquipmentType? = null,
    val numberOfSlots: List<Int>? = null,
    val rarity: List<Int>? = null,
    val gender: Gender? = null,
    val hunterType: HunterType? = null,
    val skills: List<Int>? = null,
)

File: lib/features/armor/domain/armor_filter.dart
import '../../../core/domain/enums.dart';
import '../../skill/domain/skill.dart';

class const ArmorFilter({
  final String? name,
  final EquipmentType? type,
  final List<int>? numberOfSlots,
  final List<int>? rarity,
  final Gender? gender,
  final HunterType? hunterType,
  final List<SkillTree>? skills,
});

class const ArmorSetFilter({
  final String? name,
  final List<int>? rarity,
  final Rank? rank,
  final HunterType? hunterType,
  final Gender? gender,
  final List<SkillTree>? skills,
});
